core/src/test/scala/unit/kafka/admin/ListConsumerGroupTest.scala

```diff
@@ -46,16 +46,16 @@ class ListConsumerGroupTest extends ConsumerGroupCommandTest {
     }, s"Expected --list to show groups $expectedGroups, but found $foundGroups.")
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
-  def testListWithUnrecognizedNewConsumerOption(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
+  def testListWithUnrecognizedNewConsumerOption(quorum: String, groupProtocol: String): Unit = {
     val cgcArgs = Array("--new-consumer", "--bootstrap-server", bootstrapServers(), "--list")
     assertThrows(classOf[OptionException], () => getConsumerGroupService(cgcArgs))
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
-  def testListConsumerGroupsWithStates(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersAll"))
+  def testListConsumerGroupsWithStates(quorum: String, groupProtocol: String): Unit = {
```

Comment:
Same question. Is it necessary? Or would using `@ValueSource(strings = Array("zk", "kraft", "kraft+kip-848"))` be enough?

Reply:
Hi yes thanks for bringing that up, My original idea was that the protocol would somehow dictate the type of consumer created and then we would test all the possibilities. But after a couple discussions we realized that there's no 1:1 mapping between protocol type and group type right so I still left the combinations there cause it's still better to have full coverage. I could just test with the new GC and the old GC aka what you're suggesting, what do you think?

Reply:
Let me also figure out how to create a consumer to use the protocol type specified, I wasn't able to figure it out before. I realized that classic was the default type created irrespective of what protocol type I put in